### ISSUE TITLE
L1 arrow compaction

### ISSUE THREAD

**Issue by thorfour** [polarsignals/frostdb#433]
It may be useful to have the option to compact L0 arrow records into L1 arrow records instead of Parquet.

**Comment by thorfour** [comment]
This may only be worth pursuing once the REE support changes are in FrostDB as well as the record sorting implementation https://github.com/apache/arrow/pull/34719 is completed

**Comment by asubiotto** [comment]
Agreed. I think moving to arrow-only in-mem would be the last step in this quarter.

**Comment by gernest** [comment]
I am thinking about this,  I was wondering if this is the same as `arrowutils.MergeRecords(arrow_parts...) |> arrowutils.SortRecord |> parts.NewArrowPart`  ?

Looking at `MergeRecords` got me wondering how do we handle sorting by dynamic columns ? I mean when the records have mismatch number of dynamic columns?

**Comment by asubiotto** [comment]
Yes, although given the arrow parts should be merged on input, there probably isn't a need for the downstream sort. I'd also be interested in getting some L0 to L1 stats on how much memory we reduce through arrow compaction vs parquet compaction.

**Comment by gernest** [comment]
@asubiotto can you expand a bit about memory expectation between arrow/parquet compaction ?

I was always under the impression `parquet+compression` gives better memory saving than `arrow` ?

**Comment by asubiotto** [comment]
Yes, this is why I'd be interested in getting some numbers so we are informed about the tradeoffs. Intuitively, dictionary encoding should go a long way. We've also been thinking about experimenting with run end encoding in arrow.

**Comment by asubiotto** [comment]
I think it's still useful to keep this open.
